lib/parser/parse-tree.h

```diff
@@ -3703,7 +3704,13 @@ struct OpenMPBlockConstruct {
 
 struct OpenMPLoopConstruct {
   TUPLE_CLASS_BOILERPLATE(OpenMPLoopConstruct);
-  std::tuple<OmpLoopDirective, OmpClauseList> t;
+  template<typename A, typename B, typename = common::NoLvalue<A>,
+      typename = common::NoLvalue<B>>
+  OpenMPLoopConstruct(A &&a, B &&b)
+    : t({std::move(a), std::move(b), std::nullopt, std::nullopt}) {}
```

Comment:
Why is this constructor a template? Don't `A` and `B` have to be `OmpLoopDirective` and `OmpClauseList`?